tests/e2e-test-plugin/e2e-test-plugin.php

```diff
@@ -0,0 +1,236 @@
+				'inner_content_replacements' => array(
+					array(
+						'search'  => 'wp-image-' . $source_data['id'],
+						'replace' => 'wp-image-' . $image_id,
+					),
+					array(
+						'search'  => $source_data['url'],
+						'replace' => wp_get_attachment_image_url( $image_id, 'full' ),
+					),
+				),
+			);
+		},
+	)
+);
+
+// Distributor data registration for the shortcode.
+distributor_register_data(
+	'term_shortcode_data',
+	array(
+		'location'           => 'post_content',
+		'attributes'         => array(
+			'shortcode'           => 'dt_term_shortcode',
+			'shortcode_attribute' => 'id',
+		),
+		'pre_distribute_cb'  => function( $data ) {
+			$term = get_term_by( 'id', $data, 'category' );
+			if ( ! $term ) {
+				return array();
+			}
+
+			return array(
+				'name'        => $term->name,
+				'slug'        => $term->slug,
+				'description' => $term->description,
+				'taxonomy'    => $term->taxonomy,
+			);
+		},
+		'post_distribute_cb' => function( $extra_data, $source_data, $post_data ) {
+			if ( ! isset( $extra_data['name'] ) && ! isset( $extra_data['taxonomy'] ) ) {
+				return;
+			}
+
+			$existing_term = get_term_by( 'name', $extra_data['name'], $extra_data['taxonomy'] );
+			if ( $existing_term ) {
+				return $existing_term->term_id;
+			}
+
+			// Create the term.
+			$term = wp_insert_term(
+				$extra_data['name'],
+				$extra_data['taxonomy'],
+				array(
+					'description' => $extra_data['description'],
+					'slug'        => $extra_data['slug'],
+				)
+			);
+
+			if ( is_wp_error( $term ) ) {
+				return;
+			}
+
+			return $term['term_id'];
+		},
+	)
+);
+
+/**
+ * Pre-distribute callback for the cover block data.
+ *
+ * @param mixed $image_id           Background image ID.
+ * @return array{image_url: string} Extra data to be added to the request.
+ */
+function distributor_image_pre_distribute_cb( $image_id ) {
+	if ( ! $image_id ) {
+		return array();
+	}
+
+	$image_url = wp_get_attachment_image_url( $image_id, 'full' );
+	return array(
+		'image_url' => $image_url,
+	);
+}
+
+/**
+ * Post-distribute callback for the cover block data.
+ *
+ * @param array $extra_data  Extra data related to the image.
+ * @param mixed $source_data Source data.
+ * @param array $post_data   Post data.
+ * @return bool|int
+ */
+function distributor_image_post_distribute_cb( $extra_data, $source_data, $post_data ) {
+	if ( ! isset( $extra_data['image_url'] ) ) {
+		return;
+	}
+
+	$source_image_url = $extra_data['image_url'];
+	$source_image_id  = $source_data;
+
+	// Check if the media is already existing on the site. If it is, return the media ID.
+	$image_id = get_attachment_id_by_original_data( $source_image_id, $source_image_url );
+
+	if ( ! $image_id ) {
+		$image_id = process_media( $source_image_url, 0, [] );
+		update_post_meta( $image_id, 'dt_original_media_id', $source_image_id );
+		update_post_meta( $image_id, 'dt_original_media_url', $source_image_url );
```

Comment:
Post meta needs to be slashed https://github.com/WordPress/wordpress-develop/blob/bcb59bfb14cbfa79335fc9e411aae56b060da85d/src/wp-includes/meta.php#L210-L215

```suggestion
		update_post_meta( $image_id, 'dt_original_media_id', wp_slash( $source_image_id ) );
		update_post_meta( $image_id, 'dt_original_media_url', wp_slash( $source_image_url ) );
```